configs/configgen.py

```diff
@@ -100,6 +100,8 @@ def generate_config(template_path, template, output_file, **context):
 # optional external service ports: built from external_virtual_hosts above. Each external host
 #                                  that Envoy proxies to listens on its own port.
 # optional mongo ports: built from mongos_servers above.
+# TODO(wattli): investigate why a bad config (with port but not address) can
```

Comment:
Sorry this is not an acceptable TODO, it will cause production issues.

The issue is that you are returning an std::string from the configuration, and it doesn't get bound to an address until the listeners actually start running on the workers. I mentioned this a while ago, but please have the configuration return an Address::InstancePtr instead of a string. This will cause an exception to be thrown if the address cannot be resolved at config load time.